libs/natparse/src/main/java/org/amshove/natparse/parsing/StatementListParser.java

```diff
@@ -1652,12 +1652,23 @@ private ComparisonOperator parseRelationalOperator(RelationalCriteriaNode node)
 			case NOT ->
 			{
 				consume(node);
-				consumeAnyMandatory(node, List.of(SyntaxKind.EQUALS_SIGN, SyntaxKind.EQ, SyntaxKind.EQUAL));
-				if (previousToken().kind() == SyntaxKind.EQUAL)
+				consumeAnyMandatory(node, List.of(SyntaxKind.EQUALS_SIGN, SyntaxKind.EQ, SyntaxKind.EQUAL, SyntaxKind.LESSER_SIGN, SyntaxKind.LT, SyntaxKind.GREATER_SIGN, SyntaxKind.GT));
+				switch (previousToken().kind())
 				{
-					consumeOptionally(node, SyntaxKind.TO);
+					case LT, LESSER_SIGN ->
+					{
+						yield ComparisonOperator.GREATER_OR_EQUAL;
```

Comment:
You only need the `yield` if the case has a block `{ }`.

`case LT, LESSER_SIGN -> ComparisonOperator.GREATER_OR_EQUAL;`

is the same as what you have. So SonarQube wants you to remove the `{ }` and don`t use yield, as this case only has a single expression.

Reply:
Yeah, I did try that, but got syntax errors. I’ll give it another shot tomorrow.